src/freenet/node/PeerLocation.java

```diff
@@ -56,43 +60,43 @@ public synchronized int getDegree() {
 	}
 
 	boolean updateLocation(double newLoc, double[] newLocs) {
-		if(!Location.isValid(newLoc)) {
+		if (!Location.isValid(newLoc)) {
 			Logger.error(this, "Invalid location update for " + this+ " ("+newLoc+')', new Exception("error"));
 			// Ignore it
 			return false;
 		}
 
-		for(double currentLoc : newLocs) {
-			if(!Location.isValid(currentLoc)) {
-				Logger.error(this, "Invalid location update for " + this + " ("+currentLoc+')', new Exception("error"));
+		final double[] newPeersLocation = new double[newLocs.length];
+		for (int i = 0; i < newLocs.length; i++) {
+			final double loc = newLocs[i];
+			if (!Location.isValid(loc)) {
+				Logger.error(this, "Invalid location update for " + this + " (" + loc + ")", new Exception("error"));
 				// Ignore it
 				return false;
```

Comment:
In cases where the target is between two sequential elements of the FOAF list, yes. When the target is between the last and the first (wrapping around), the resulting location may be different (but the distance equal). My comment to your other node explains this more in-depth.